Ensure num_threads is initialized before calling omp_get_max_threads

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60185 Ensure num_threads is initialized before calling omp_get_max_threads**
* #60184 OpenMP: Refactor parallel_reduce to share code with parallel_for
* #60183 Ensure thread id is valid in nested parallel regions

`get_num_threads` is usually called before `parallel_for` so there's no
guaruntee we've initialized `num_threads` properly.

Differential Revision: [D29287814](https://our.internmc.facebook.com/intern/diff/D29287814)

peterbell10 committed:
Ensure num_threads is initialized before calling omp_get_max_threads

`get_num_threads` is usually called before `parallel_for` so there's no
guaruntee we've initialized `num_threads` properly.

[ghstack-poisoned]

peterbell10 committed:
Update on "Ensure num_threads is initialized before calling omp_get_max_threads"


`get_num_threads` is usually called before `parallel_for` so there's no
guaruntee we've initialized `num_threads` properly.

[ghstack-poisoned]

peterbell10 committed:
Update on "Ensure num_threads is initialized before calling omp_get_max_threads"


`get_num_threads` is usually called before `parallel_for` so there's no
guaruntee we've initialized `num_threads` properly.

[ghstack-poisoned]

peterbell10 committed:
Update on "Ensure num_threads is initialized before calling omp_get_max_threads"


`get_num_threads` is usually called before `parallel_for` so there's no
guaruntee we've initialized `num_threads` properly.

[ghstack-poisoned]

peterbell10 committed:
Update on "Ensure num_threads is initialized before calling omp_get_max_threads"


`get_num_threads` is usually called before `parallel_for` so there's no
guaruntee we've initialized `num_threads` properly.

[ghstack-poisoned]

peterbell10 committed:
Update on "Ensure num_threads is initialized before calling omp_get_max_threads"


`get_num_threads` is usually called before `parallel_for` so there's no
guaruntee we've initialized `num_threads` properly.

[ghstack-poisoned]

peterbell10 committed:
Update on "Ensure num_threads is initialized before calling omp_get_max_threads"


`get_num_threads` is usually called before `parallel_for` so there's no
guaruntee we've initialized `num_threads` properly.

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29287814).

peterbell10 committed:
Update on "Ensure num_threads is initialized before calling omp_get_max_threads"


`get_num_threads` is usually called before `parallel_for` so there's no
guaruntee we've initialized `num_threads` properly.

Differential Revision: [D29287814](https://our.internmc.facebook.com/intern/diff/D29287814)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29287814).
